ddspipe_yaml/src/cpp/YamlValidator.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file YamlValidator.cpp
+ *
+ */
+
+#include <cpp_utils/Log.hpp>
+#include <ddspipe_yaml/YamlValidator.hpp>
+
+namespace eprosima {
+namespace ddspipe {
+namespace yaml {
+
+bool YamlValidator::validate_tags(
+        const Yaml& yml,
+        const std::set<TagType>& tags)
+{
+    if (!yml.IsMap() && !yml.IsNull())
```

Comment:
You should take into account also the error of adding a tag that is supposed to have tags inside being empty. An example of this error would be writing:
recorder:
..output:
..compression:
....algorithm: none
In the previous example the tag ``output`` should have some other tag inside such as filename, path, ...

Reply:
I believe that configuration file is valid. All the tags inside `output` are optional, so its not a problem that none of them are present.